hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/WrapperAsyncFSOutput.java

```diff
@@ -91,7 +93,11 @@ private void flush0(CompletableFuture<Long> future, ByteArrayOutputStream buffer
           out.hflush();
         }
       }
-      future.complete(out.getPos());
+      long pos = out.getPos();
+      if(pos > this.syncedLength) {
```

Comment:
I had asked in #1970 if this can be implemented with an `AtomicLong` instead. @Apache9 responded with an analysis and alternative suggestion. Following his suggestion, it would be nice to add a comment on `WrapperAsyncFSOutput` stating that we expect an instance to be used only from a single thread, foregoing synchronization for the sake of performance.

Reply:
Yes, I agree with you, and even the 'if(pos > this.syncedLength) {' could be removed as @Apache9  suggested, but the current PR is backport ,so we'd better keep it in sync with the one applied to master.After this PR is applied to 2.x, I would open a addnum PR for both master and 2.x to incorporate your suggestions.